File: M3SDK_Android/M3SDK_Demo/UGR_Java/app/src/main/java/net/m3mobile/ugr_demo/UhfTag.java
package net.m3mobile.ugr_demo;

import android.annotation.SuppressLint;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by M3 on 2017-12-07.
 */

public class UHFTag {
    public String EPC;
    public int Reads;
    public String TIME;
    public long time, diff;

    public UHFTag(String EPC, int reads) {
        this.EPC = EPC;
        Reads = reads;

        long currentTime = System.currentTimeMillis();
        time = currentTime;
        @SuppressLint("SimpleDateFormat") SimpleDateFormat currentDayTime = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
        TIME = currentDayTime.format(new Date(currentTime));
    }
}

File: M3SDK_Android/M3SDK_Demo/UGR_kotlin/app/src/main/java/net/m3mobile/ugr_demo/UHFTag.kt
package net.m3mobile.ugr_demo

import android.annotation.SuppressLint
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by M3 on 2017-12-07.
 */
class UHFTag(var EPC: String, var Reads: Int) {
    var TIME: String
    var time: Long
    var diff: Long = 0

    init {
        val currentTime = System.currentTimeMillis()
        time = currentTime
        @SuppressLint("SimpleDateFormat") val currentDayTime =
            SimpleDateFormat("yyyy-MM-dd hh:mm:ss")
        TIME = currentDayTime.format(Date(currentTime))
    }
}